src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -240,3 +277,30 @@ void QgsExternalResourceConfigDlg::setConfig( const QVariantMap &config )
     }
   }
 }
+
+QgsExpressionContext QgsExternalResourceConfigDlg::createExpressionContext() const
+{
+  QgsExpressionContext context = QgsEditorConfigWidget::createExpressionContext();
+  context << QgsExpressionContextUtils::formScope( );
+  context << QgsExpressionContextUtils::parentFormScope( );
+
+  QgsExpressionContextScope *fileWidgetScope = QgsExternalStorageFileWidget::createFileWidgetScope();
+  context << fileWidgetScope;
+
+  context.setHighlightedVariables( fileWidgetScope->variableNames() );
+  return context;
+}
+
+void QgsExternalResourceConfigDlg::changeStorageType( int storageType )
+{
+  // first one in combo box is not an external storage
+  mExternalStorageGroupBox->setVisible( storageType );
```

Comment:
for the sake of clarity
```suggestion
  mExternalStorageGroupBox->setVisible( storageTypeIndex > 0 );
```

Reply:
Done